average.js

```diff
@@ -1,6 +1,17 @@
 
 function average(numbers) {
-  return numbers.reduce((p, c)=> p + c, 0) / numbers.length;
+  
+  let invalidData = 0;
+  
+  return parseFloat(numbers.reduce((p, c)=> {
+    if (isNaN(c)){
+      invalidData++ ;
+      return p + 0;  
+    } else {
+      return p + c;
+    }
+  }, 0)) / (numbers.length - invalidData);
+  
 }
 
-module.exports = {average};
+module.exports = { average };
```

Comment:
Try a new test: The tests in `test/average.test.js` express the functionality of the `average` function.
Consider this situation: In IoT applications, analog sensors will report occasional outliers. 'Outliers' are values so small or so large, they aren't expected to happen. We need new functionality to ignore these outliers.
First add a unit-test in `test/average.test.js` to supply outliers and expect them to be ignored. Name the test appropriately. Then implement it in the code.
You'll need thresholds to recognize outliers: Think of sensing battery-temperature values and determine reasonable values.

Reply:
Hello @numocityadmin @sudeeprp I have added the outliers test in test/average.test.js . I used an article named [Ways to Detect and Remove the Outliers](https://towardsdatascience.com/ways-to-detect-and-remove-the-outliers-404d16608dba) as the reference to find the lower and upper bounds using the interquartile range (IQR), also called the midspread. I have tested using some values and it has passed the test. 

please share your reviews and comments for this approach and code, thank you